### PR TITLE
feat(billing): add generic BillingUsage model for quota tracking

### LINKED ISSUE #3269: feat(billing): generic BillingUsage model for quota tracking
## Context
Downstream projects (Lou, etc.) need per-plan quotas. The stack should provide a generic usage tracking model.

## Spec
- New model `BillingUsage { organizationId, month (YYYY-MM), counters: Object }` 
- `counters` is a free-form object — each project defines its own keys (e.g. `{ executions: 42, aiCalls: 5 }`)
- Atomic `$inc` on counters (no read-then-write race conditions)
- Auto-create on first increment
- Lazy monthly reset (or check month on read)
- Service: `increment(orgId, key, amount)`, `get(orgId)`, `reset(orgId)`

## Acceptance
- Model + repository + service + tests
- No project-specific logic — purely generic

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

- Add `BillingUsage` Mongoose model with `organizationId`, `month` (YYYY-MM), and free-form `counters` (Mixed), with compound unique index on `{ organizationId, month }`
- Add repository with `get`, `increment` (atomic `$inc` with upsert), and `reset` operations
- Add service layer that auto-computes current month and delegates to repository
- Add Zod validation schema for `BillingUsage`
- Add 11 unit tests covering schema validation, upsert creation, atomic increment, empty counters fallback, and reset

Closes #3269

## Test plan

- [x] `npm run lint` passes (0 errors)
- [x] All 227 unit tests pass (14 suites)
- [x] New `billing.usage.unit.tests.js` — 11 tests covering schema + service layer

## Summary by CodeRabbit

* **New Features**
  * Added billing usage tracking functionality to monitor organizational consumption metrics on a monthly basis.
  * Enabled retrieval, increment, and reset operations for usage counters.

* **Tests**
  * Added comprehensive unit tests for billing usage validation and tracking operations.

### CODE REVIEW

File: modules/billing/repositories/billing.usage.repository.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @param {String} key - The counter key to increment (e.g. 'executions').
+ * @param {Number} amount - The amount to increment by.
+ * @returns {Promise<Object>} The updated usage document.
+ */
+const increment = (organizationId, month, key, amount) => BillingUsage.findOneAndUpdate(
+  { organizationId, month },
+  { $inc: { [`counters.${key}`]: amount } },
+  { upsert: true, returnDocument: 'after', runValidators: true },
+).exec();
+
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/billing/repositories/billing.usage.repository.js]: Fixed in fc34ae8a: Added E11000 duplicate key catch with retry on concurrent upserts
